src/main/java/com/yugabyte/sample/common/metrics/PromMetrics.java

```diff
@@ -0,0 +1,144 @@
+                        return null;
+                    }
+                    public void checkClientTrusted(X509Certificate[] certs, String authType) {
+                    }
+                    public void checkServerTrusted(X509Certificate[] certs, String authType) {
+                    }
+                }
+            };
+
+            SSLContext sc = SSLContext.getInstance("TLS");
+            sc.init(null, trustAllCerts, new java.security.SecureRandom());
+            HttpsURLConnection.setDefaultSSLSocketFactory(sc.getSocketFactory());
+
+            HostnameVerifier allHostsValid = (hostname, session) -> true;
+            HttpsURLConnection.setDefaultHostnameVerifier(allHostsValid);
+        } catch (Exception e) {
+            throw new IOException("Failed to disable SSL verification", e);
+        }
+    }
+
+    /*
+     * Fetches the counter for the given metric and table name
+     * and accumulates across all the T-Servers.
+     */
+    public long getCounter(String metricName, String tableName) {
+        long counter = 0;
+        for (String promContactPoint : promContactPoints) {
+            long fetchedCounter = fetchPromCounter(metricName, tableName, promContactPoint);
+            if (fetchedCounter > 0) {
+                counter += fetchedCounter;
+            }
+        }
+
+        return counter;
+    }
+
+    /*
+     * Fetches the metric counter for one T-Server.
+     */
+    private long fetchPromCounter(String metricName, String tableName, String promContactPoint) {
+        HttpURLConnection connection = null;
+
+        try {
+            URL url = new URL(promContactPoint);
+            connection = (HttpURLConnection) url.openConnection();
+            connection.setRequestMethod("GET");
+
+            int responseCode = connection.getResponseCode();
+            if (responseCode == HttpURLConnection.HTTP_OK) {
+                BufferedReader reader = new BufferedReader(
+                    new InputStreamReader(connection.getInputStream()));
+                StringBuilder response = new StringBuilder();
+                String line;
+
+                while ((line = reader.readLine()) != null) {
+                    response.append(line).append("\n");
+                }
+                reader.close();
+
+                // Example: match restart_read_requests{table_name="usertable", ...} 10 1234567890
+                //                                                                    ^
+                //                                                                    |
+                //                                                                 counter
+                Pattern pattern = Pattern.compile(
+                    "^" + metricName + "\\{[^}]*table_name=\"" + tableName +
+                    "\"[^}]*\\}\\s+(\\d+)\\s+(\\d+)", Pattern.MULTILINE);
+                Matcher matcher = pattern.matcher(response.toString());
+
+                if (matcher.find()) {
```

Comment:
can we assert that there is only 1 pattern match for the table and metric that we have?